tests/compiletest.rs

```diff
@@ -1,71 +1,432 @@
+    if actual == expected {
+        return;
+    }
+    eprintln!("actual output differed from expected {}", path.display());
+    eprintln!("{}", pretty_assertions::StrComparison::new(expected, actual));
+    eprintln!()
+}
+
+fn ignore_file(p: &Path, target: &str) -> bool {
+    let content = std::fs::read_to_string(p).unwrap();
+    for line in content.lines() {
+        if let Some(s) = line.strip_prefix("// ignore-") {
+            let s =
+                s.split_once(|c: char| c == ':' || c.is_whitespace()).map(|(s, _)| s).unwrap_or(s);
+            if target.contains(s) {
+                return true;
+            }
+            if get_pointer_width(target).contains(s) {
+                return true;
+            }
+        }
+        if let Some(s) = line.strip_prefix("// only-") {
+            let s =
+                s.split_once(|c: char| c == ':' || c.is_whitespace()).map(|(s, _)| s).unwrap_or(s);
+            if !target.contains(s) {
+                return true;
+            }
+            if !get_pointer_width(target).contains(s) {
+                return true;
+            }
+        }
+    }
+    false
+}
+
+// Taken 1:1 from compiletest-rs
+fn get_pointer_width(triple: &str) -> &'static str {
+    if (triple.contains("64") && !triple.ends_with("gnux32") && !triple.ends_with("gnu_ilp32"))
+        || triple.starts_with("s390x")
+    {
+        "64bit"
+    } else if triple.starts_with("avr") {
+        "16bit"
+    } else {
+        "32bit"
+    }
+}
+
+fn extract_env(cmd: &mut Command, path: &Path) {
+    let content = std::fs::read_to_string(path).unwrap();
+    for line in content.lines() {
+        if let Some(s) = line.strip_prefix("// compile-flags:") {
+            cmd.args(s.split_whitespace());
+        }
+        if let Some(s) = line.strip_prefix("// rustc-env:") {
+            for env in s.split_whitespace() {
+                if let Some((k, v)) = env.split_once('=') {
+                    cmd.env(k, v);
+                }
+            }
+        }
+    }
+}
 
-    run_tests("compile-fail", path, target);
+macro_rules! regexes {
+    ($($regex:expr => $replacement:expr,)*) => {lazy_static::lazy_static! {
+        static ref REGEXES: Vec<(Regex, &'static str)> = vec![
+            $((Regex::new($regex).unwrap(), $replacement),)*
+        ];
+    }};
 }
 
-fn miri_pass(path: &str, target: &str) {
+regexes! {
+    // erase line and column info
+    "\\.rs:[0-9]+:[0-9]+"            => ".rs:LL:CC",
+    // erase alloc ids
+    "alloc[0-9]+"                    => "ALLOC",
+    // erase borrow stack indices
+    "<[0-9]+>"                       => "<BORROW_IDX>",
+    // erase whitespace that differs between platforms
+    " +at (.*\\.rs)"                 => " at $1",
+    // erase generics in backtraces
+    "([0-9]+: .*)::<.*>"             => "$1",
+    // erase addresses in backtraces
+    "([0-9]+: ) +0x[0-9a-f]+ - (.*)" => "$1$2",
+    // erase hexadecimals
+    "0x[0-9a-fA-F]+(\\[a[0-9]+\\])?" => "$$HEX",
```

Comment:
Hard to distinguish and not always true with alignment and such

Reply:
Can you add a FIXME and an issue (or make it part of a larger tracking issue) to make this more targeted? It is a shame to no longer test the precise offsets/values that we used to test.

Reply:
I set it up so that the stderr output does not contain this, but the error patterns work on the unnormalized stderr, giving us back a way to test these

Reply:
I think we can maybe also do something where only hex numbers at least 4 digits long are normalized? That would still cover "invalid pointer 0xdaf130", but would allow "error at offset 0x10" to persist.